hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenDissociateFromAccountHandler.java

```diff
@@ -45,13 +70,192 @@ public TokenDissociateFromAccountHandler() {
+                    numNftsToSubtract += tokenRelBalance;
+                }
+            } else {
+                // Handle active tokens
+                validateFalse(tokenRel.accountNumber() == token.treasuryAccountNumber(), ACCOUNT_IS_TREASURY);
+                validateFalse(tokenRel.frozen(), ACCOUNT_FROZEN_FOR_TOKEN);
+
+                if (tokenRelBalance > 0) {
+                    validateFalse(token.tokenType() == TokenType.NON_FUNGIBLE_UNIQUE, ACCOUNT_STILL_OWNS_NFTS);
+                    // If the fungible token is NOT expired, then we throw an exception because we
+                    // can only dissociate tokens with a zero balance by this time in the code
+                    // @future('6864'): uncomment when token expiry is implemented
+                    // validateTrue(tokenIsExpired, TRANSACTION_REQUIRES_ZERO_TOKEN_BALANCES);
+
+                    // If the fungible common token is expired, we automatically transfer the
+                    // dissociating account's balance back to the token's treasury
+                    final var treasuryTokenRel = dissociation.treasuryTokenRel();
+                    if (treasuryTokenRel != null) {
+                        final var updatedTreasuryBalanceTokenRel = treasuryTokenRel.balance() + tokenRelBalance;
+                        treasuryBalancesToUpdate.add(treasuryTokenRel
+                                .copyBuilder()
+                                .balance(updatedTreasuryBalanceTokenRel)
+                                .build());
+                    }
+                }
+            }
+
+            // Regardless of the token status, now the token relation with the given account ID must be updated to zero
+            final var updatedBalanceTokenRel =
+                    tokenRel.copyBuilder().balance(0L).build();
+            tokenRelsToRemove.add(updatedBalanceTokenRel);
+
+            if (tokenRel.automaticAssociation()) {
+                numAutoAssociationsToSubtract++;
+            }
+
+            numAssociationsToSubtract++;
+        }
+
+        // get changes to account and token relations
+        final var updatedTokenRels =
+                new TokenRelListCalculator(tokenRelStore).removeTokenRels(account, tokenRelsToRemove);
+        final var newHeadTokenId = updatedTokenRels.updatedHeadTokenId() != null
+                ? updatedTokenRels.updatedHeadTokenId()
+                : account.headTokenNumber();
+
+        // Update the account with the aggregate number of NFTs, auto associations, and associations to remove,
+        // as well as the new head token number
+        final var updatedAcct = account.copyBuilder()
+                .numberOwnedNfts(account.numberOwnedNfts() - numNftsToSubtract)
+                .usedAutoAssociations(account.usedAutoAssociations() - numAutoAssociationsToSubtract)
+                .numberAssociations(account.numberAssociations() - numAssociationsToSubtract)
```

Comment:
This could actually be missing in `mono-service` as well; but if we dissociated from an expired token and erased a non-zero balance, we need to decrement [`numberPositiveBalances`](https://github.com/hashgraph/hedera-protobufs/blob/add-pbj-types-for-state/services/state/token/account.proto#L128) here too. (An account isn't allowed be deleted when it has non-zero token balances, so failing to decrement that field could make an account un-deletable.)

Reply:
Looks like [it is there](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/models/Account.java#L261) in mono-service. I just missed it. Good catch!

Reply:
Done